evalml/tests/automl_tests/test_automl.py

```diff
@@ -4933,3 +4933,18 @@ def test_pipeline_parameter_warnings_component_graphs(
     assert len(w) == (1 if len(set_values) else 0)
     if len(w):
         assert w[0].message.components == set_values
+
+
+@pytest.mark.parametrize("nans", [None, pd.NA, np.nan])
+def test_search_with_text_nans(nans):
+    X = pd.DataFrame(
+        {
+            "a": [np.nan] + [i for i in range(99)],
+            "b": [np.nan] + [f"string {i} is valid" for i in range(99)],
+        }
+    )
+    X.ww.init(logical_types={"b": "NaturalLanguage"})
+    y = pd.Series([0] * 25 + [1] * 75)
+
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type="binary", max_iterations=2)
```

Comment:
I think the follow-up bug that we encountered was that the types were being set to categorical by woodwork and only the LightGBMClassifier failed because of it right?. 

I think it may be better to use the `AutoMLSearchEnv` and check all of the woodwork types are only Double/Integer in `mock_fit`. I see we also check that in `test_text_featurizer.py` now but if we're going to add an AutoMLSearch test, I think it's worth it to run a full batch and check what we're after with mocks. Correct me if I'm wrong but this test wouldn't catch the LGBM problem right?

Reply:
Yeah, good point! I'll patch that